### PR TITLE
Moving to core list

### PR DESCRIPTION
**Pull request by MitchelPaulin**
Switched to d2l-list from dnd-sortable for the re-ordering capabilities. This included some required changes in d2l-list [see here](https://github.com/BrightspaceUI/core/pull/2029).

Note: v-diff will fail until d2l-list is updated

### CODE REVIEW

File: src/components/editor/rubric-criteria-group-editor.js [hunk 1]
```diff
@@ -177,56 +186,56 @@ class RubricCriteriaGroupEditor extends composeMixins(
...
+								?is-holistic="${this.isHolistic}"
+								?is-reordered="${this.isReordered}"
+								?updating-levels="${this.updatingLevels}"
+								@d2l-criteria-count-changed="${this._updateCriterionCount}"
+							>
+							</d2l-rubric-criteria-editor>
+						</div>
 					</div>
 				</div>
-			</d2l-scroll-wrapper>
```
[the remark this answers is not in the document]
Comment (MitchelPaulin) [reply src/components/editor/rubric-criteria-group-editor.js:229]: Update: We do need the functionality that d2l-scroll-wrapper provides. Will look into adding this back as part of another ticket.

File: src/components/editor/rubric-criteria-editor.js [hunk 2]
```diff
@@ -322,6 +268,8 @@ class RubricCriteriaEditor extends composeMixins(
 				composed: true,
 			}));
 		}
+
+		this._criteriaEntities = this._criteriaEntities.map((val, index) => { return { key: index.toString(), content: val }; });
```
Comment (jesseag) [inline src/components/editor/rubric-criteria-editor.js:272]: Do we know that entity.criterionHrefs will always be defined on Line 238?
Line 238 this._criteriaEntities = entity.criterionHrefs;

If it is not always defined then would we need an optional operator here similar to line 237? If entity.Hrefs is always defined then this comment doesn't really apply.
this._criteriaEntities = this._criteriaEntities?.map((val, index) => { return { key: index.toString(), content: val }; });

Comment (MitchelPaulin) [reply src/components/editor/rubric-criteria-editor.js:272]: A rubric should always have at least one criterion

Comment (jesseag) [reply src/components/editor/rubric-criteria-editor.js:272]: That makes sense.

**Review by jesseag — APPROVED** [review]
Looks okay to me. I don't know if you want to wait for a review from dbatiste first.

**Review by jesseag — APPROVED** [review]

**Comment by devpow112** [comment]
@MitchelPaulin You'll need tp update the package-lock.json file to get the newer version of `core` that was released with your changes

**Comment by MitchelPaulin** [comment]
The placement of the handle in `d2l-list` caused some elements to shift to the right, updating the goldens

**Review by jesseag — APPROVED** [review]

**Pull request merged**